packages/backend/src/open_payments/auth/middleware.ts

```diff
@@ -64,7 +65,9 @@ function toOpenPaymentsAccess(
   } as AccessItem
 }
 
-export function createTokenIntrospectionMiddleware({
+export function createTokenIntrospectionMiddleware<
+  T extends IntrospectionContext = WalletAddressUrlContext
```

Comment:
Can pass in the type, defaulting to the previous type (`WalletAddressUrlContext`). Existing routes are unchanged, but this new outgoing payment spent amount which uses token introspection and doesnt have the wallet address url can pass in it's own context without the url.